nats.go

```diff
@@ -3187,15 +3194,15 @@ func (nc *Conn) respToken(respInbox string) string {
 // Subscribe will express interest in the given subject. The subject
 // can have wildcards (partial:*, full:>). Messages will be delivered
 // to the associated MsgHandler.
-func (nc *Conn) Subscribe(subj string, cb MsgHandler) (*Subscription, error) {
-	return nc.subscribe(subj, _EMPTY_, cb, nil, false)
+func (nc *Conn) Subscribe(subj string, cb MsgHandler, opts ...ConsumerOption) (*Subscription, error) {
+	return nc.subscribe(subj, _EMPTY_, cb, nil, false, opts...)
 }
 
 // ChanSubscribe will express interest in the given subject and place
 // all messages received on the channel.
 // You should not close the channel until sub.Unsubscribe() has been called.
-func (nc *Conn) ChanSubscribe(subj string, ch chan *Msg) (*Subscription, error) {
-	return nc.subscribe(subj, _EMPTY_, nil, ch, false)
+func (nc *Conn) ChanSubscribe(subj string, ch chan *Msg, opts ...ConsumerOption) (*Subscription, error) {
```

Comment:
Should we move ConsumerOption to just SubscribeOption and possibly use it to overload stuff (options) we already have?

Reply:
It was `SubscribeOption`, you had me change it to this.

Reply:
I was just wondering if we could use the options for generic stuff too like timeouts vs what we have now.

Reply:
Yes, thats why it was more generically named before :) I will change back